core/ecschema-metadata/src/test/Metadata/Constant.test.ts

```diff
@@ -332,6 +332,99 @@ describe("Constant", () => {
+        label: "Test Constant",
+        description: "testing a constant",
+        phenomenon: "TestSchema.TestPhenomenon",
+        definition: "PI",
+        numerator: 5.5,
+      });
+
+      const ecSchema = await Schema.fromJson(schemaJson, new SchemaContext());
+      assert.isDefined(ecSchema);
+      const testItem = await ecSchema.getItem<Constant>("TestConstant");
+      assert.isDefined(testItem);
+      assert.isTrue(testItem?.schemaItemType === SchemaItemType.Constant);
+      const testConst: Constant = testItem as Constant;
+      assert.isDefined(testConst);
+      const constantSerialization = testConst.toJSON(true, true);
+
+      expect(constantSerialization.$schema).eql("https://dev.bentley.com/json_schemas/ec/32/schemaitem");
+      expect(constantSerialization.name).eql("TestConstant");
+      expect(constantSerialization.schemaVersion).eql("01.02.03");
+      expect(constantSerialization.schema).eql("TestSchema");
+
+      expect(constantSerialization.label).eql("Test Constant");
+      expect(constantSerialization.description).eql("testing a constant");
+
+      assert.strictEqual(constantSerialization.definition, "PI");
+      assert.strictEqual(constantSerialization.phenomenon, "TestSchema.TestPhenomenon");
+
+      expect(constantSerialization.numerator).eql(5.5);
+      expect(constantSerialization.denominator).to.be.undefined;
+    });
+
+    it("async - when only denominator property is defined", async () => {
+      const schemaJson = createSchemaJson({
+        label: "Test Constant",
+        description: "testing a constant",
+        phenomenon: "TestSchema.TestPhenomenon",
+        definition: "PI",
+        denominator: 5.1,
+      });
+
+      const ecSchema = await Schema.fromJson(schemaJson, new SchemaContext());
+      assert.isDefined(ecSchema);
+      const testItem = await ecSchema.getItem<Constant>("TestConstant");
+      assert.isDefined(testItem);
+      assert.isTrue(testItem?.schemaItemType === SchemaItemType.Constant);
+      const testConst: Constant = testItem as Constant;
+      assert.isDefined(testConst);
+      const constantSerialization = testConst.toJSON(true, true);
+
+      expect(constantSerialization.$schema).eql("https://dev.bentley.com/json_schemas/ec/32/schemaitem");
+      expect(constantSerialization.name).eql("TestConstant");
+      expect(constantSerialization.schemaVersion).eql("01.02.03");
+      expect(constantSerialization.schema).eql("TestSchema");
+
+      expect(constantSerialization.label).eql("Test Constant");
+      expect(constantSerialization.description).eql("testing a constant");
+
+      assert.strictEqual(constantSerialization.definition, "PI");
+      assert.strictEqual(constantSerialization.phenomenon, "TestSchema.TestPhenomenon");
+
+      expect(constantSerialization.numerator).to.be.undefined;
+      expect(constantSerialization.denominator).eql(5.1);
+    });
+
+    it("sync - when numerator and denominator properties are undefined", () => {
```

Comment:
do we really need all three of these tests?  What additional cases are we testing by having all three instead of just the first two or just the last one?

Reply:
Consider this feedback on all the tests

Reply:
Extra test cases are removed.